packages/web3-types/src/web3_base_provider.ts

```diff
@@ -159,27 +162,32 @@ export abstract class Web3BaseProvider<API extends Web3APISpec = EthExecutionAPI
 
 	public abstract on(
 		type: 'disconnect',
-		callback: Web3ProviderEventCallback<ProviderRpcError>,
+		listener: Web3Eip1193ProviderEventCallback<ProviderRpcError>,
 	): void;
 	public abstract on<T = JsonRpcResult>(
 		type: 'message' | string,
-		callback: Web3ProviderEventCallback<T>,
+		listener: Web3Eip1193ProviderEventCallback<ProviderMessage> | Web3ProviderEventCallback<T>,
+	): void;
+	public abstract on(
+		type: 'connect',
+		listener: Web3Eip1193ProviderEventCallback<ProviderConnectInfo>,
 	): void;
 	public abstract on(
-		type: 'connect' | 'chainChanged',
-		callback: Web3ProviderEventCallback<ProviderConnectInfo>,
+		type: 'chainChanged',
+		listener: Web3Eip1193ProviderEventCallback<string>,
 	): void;
 	public abstract on(
 		type: 'accountsChanged',
-		callback: Web3ProviderEventCallback<{
-			readonly accounts: string[];
-		}>,
+		listener: Web3Eip1193ProviderEventCallback<string[]>,
+	): void;
+	public abstract removeListener(
+		type: string,
+		listener: Web3Eip1193ProviderEventCallback<any> | Web3ProviderEventCallback,
 	): void;
-	public abstract removeListener(type: string, callback: Web3ProviderEventCallback): void;
 
-	public abstract once?<T = JsonRpcResult>(
+	public abstract once?<T>(
 		type: string,
-		callback: Web3ProviderEventCallback<T>,
+		callback: Web3Eip1193ProviderEventCallback<T> | Web3ProviderEventCallback<T>,
 	): void;
```

Comment:
I think `removeListener` and `once` should have the same type restrictions as `on`.